.ci/magic-modules/branch-magic-modules.sh

```diff
@@ -21,4 +21,13 @@ git checkout -B "$BRANCH"
 # this output is no longer available.
 echo "$BRANCH" > ./branchname
 
+set +x
+# Don't show the credential in the output.
+echo "$CREDS" > ~/github_private_key
+set -x
+chmod 400 ~/github_private_key
+
+# Update to head on master on all submodules, so we avoid spurious diffs.
+ssh-agent bash -c "ssh-add ~/github_private_key; git submodule update --remote --init build/terraform build/puppet/sql build/puppet/compute"
```

Comment:
Do you need creds to do this since it only fetches master HEAD from public repo?

Reply:
Unfortunately yes - because it's an ssh connection to do the fetch, you need to have a private key which is associated with _some_ github account.  It can be any account, but this is the one we've got sitting around.

The other option would be to do HTTPS fetches, but switching over to that is kind of a hassle.  What do you think?

Reply:
Totally fine the way it is. I was just curious